mapper/tests.py

```diff
@@ -344,10 +352,269 @@ def test_form_correct_submission(self):
+            for i in range(20):
+                cursor.execute("INSERT INTO table1 VALUES (%s)", [str(i)])
+
+        migrate_subscriptions.CHUNK_SIZE = 10
+        # Queries:
+        # 1. Start transaction/savepoint
+        # 2. Create cursor
+        # 3. Fetch first batch
+        # 4. Fetch second batch
+        # 5. Fetch third (empty) batch
+        # 6. End transaction/savepoint
+        with self.assertNumQueries(6, using='identities'):
+            identities = sorted(
+                migrate_subscriptions.fetch_identities(migrate))
+        self.assertEqual(identities, sorted([str(i) for i in range(20)]))
+
+    @mock.patch('mapper.tasks.MigrateSubscriptionsTask.migrate_identity')
+    @mock.patch('mapper.tasks.MigrateSubscriptionsTask.fetch_identities')
+    @mock.patch('mapper.tasks.MigrateSubscriptionsTask.count_identities')
+    def test_run(self, count_identities, fetch_identities, migrate_identity):
+        """
+        Running the task should call the various functions with correct
+        parameters.
+        """
+        migrate = MigrateSubscription.objects.create(
+            from_messageset=1, to_messageset=2,
+            table_name='table1', column_name='column1',
+        )
+
+        count_identities.return_value = 2
+        fetch_identities.return_value = ['identity1', 'identity2']
+
+        migrate_subscriptions.delay(migrate.pk)
+
+        count_identities.assert_called_once_with(migrate)
+        fetch_identities.assert_called_once_with(migrate)
+        self.assertEqual(migrate_identity.call_count, 2)
+        migrate_identity.assert_any_call(migrate, 'identity1')
+        migrate_identity.assert_any_call(migrate, 'identity2')
+
+        migrate.refresh_from_db()
+        self.assertEqual(migrate.status, MigrateSubscription.COMPLETE)
+        self.assertNotEqual(migrate.task_id, None)
+        self.assertNotEqual(migrate.completed_at, None)
+        self.assertEqual(migrate.total, 2)
+        self.assertEqual(migrate.current, 2)
+
+    @mock.patch('mapper.tasks.MigrateSubscriptionsTask.log')
+    @mock.patch('mapper.tasks.MigrateSubscriptionsTask.count_identities')
+    def test_run_failure_args(self, count_identities, log):
+        """
+        If the task raises an exception, and the migrateion object was provided
+        in the args, it should create a log object for it and log it.
+        """
+        migrate = MigrateSubscription.objects.create(
+            from_messageset=1, to_messageset=2,
+            table_name='table1', column_name='column1',
+        )
+
+        def error_effect(migrate):
+            raise Exception('Test error')
+        count_identities.side_effect = error_effect
+
+        migrate_subscriptions.delay(migrate.pk)
+
+        log_args = log.call_args[0]
+        self.assertEqual(log_args[0], migrate)
+        self.assertEqual(log_args[1], logging.ERROR)
+        self.assertTrue('Exception' in log_args[2])
+        self.assertTrue('Test error' in log_args[2])
```

Comment:
Shouldn't we check the MigrateSubscription objects status field was updated as well?